src/interface.jl

```diff
@@ -0,0 +1,709 @@
+∘(b1::Bijector, b2::Bijector) = composel(b2, b1)
+
+inv(ct::Composed) = Composed(map(inv, reverse(ct.ts)))
+
+# # TODO: should arrays also be using recursive implementation instead?
+function (cb::Composed{<:AbstractArray{<:Bijector}})(x)
+    res = x
+    for b ∈ cb.ts
+        res = b(res)
+    end
+
+    return res
+end
+
+# recursive implementation like this allows type-inference
+_transform(x, b1::Bijector, b2::Bijector) = b2(b1(x))
+_transform(x, b::Bijector, bs::Bijector...) = _transform(b(x), bs...)
+(cb::Composed{<:Tuple})(x) = _transform(x, cb.ts...)
+
+function _logabsdetjac(x, b1::Bijector, b2::Bijector)
+    res = forward(b1, x)
+    return logabsdetjac(b2, res.rv) .+ res.logabsdetjac
+end
+function _logabsdetjac(x, b1::Bijector, bs::Bijector...)
+    res = forward(b1, x)
+    return _logabsdetjac(res.rv, bs...) .+ res.logabsdetjac
+end
+logabsdetjac(cb::Composed, x) = _logabsdetjac(x, cb.ts...)
+
+# Recursive implementation of `forward`
+# NOTE: we need this one in the case where `length(cb.ts) == 2`
+# in which case forward(...) immediately calls `_forward(::NamedTuple, b::Bijector)`
+function _forward(f::NamedTuple, b::Bijector)
+    y, logjac = forward(b, f.rv)
+    return (rv=y, logabsdetjac=logjac .+ f.logabsdetjac)
+end
+function _forward(f::NamedTuple, b1::Bijector, b2::Bijector)
+    f1 = forward(b1, f.rv)
+    f2 = forward(b2, f1.rv)
+    return (rv=f2.rv, logabsdetjac=f2.logabsdetjac .+ f1.logabsdetjac .+ f.logabsdetjac)
+end
+function _forward(f::NamedTuple, b::Bijector, bs::Bijector...)
+    f1 = forward(b, f.rv)
+    f_ = (rv=f1.rv, logabsdetjac=f1.logabsdetjac .+ f.logabsdetjac)
+    return _forward(f_, bs...)
+end
+_forward(x, b::Bijector, bs::Bijector...) = _forward(forward(b, x), bs...)
+forward(cb::Composed{<:Tuple}, x) = _forward(x, cb.ts...)
+
+function forward(cb::Composed, x)
+    rv, logjac = forward(cb.ts[1], x)
+    
+    for t in cb.ts[2:end]
+        res = forward(t, rv)
+        rv = res.rv
+        logjac = res.logabsdetjac .+ logjac
+    end
+    return (rv=rv, logabsdetjac=logjac)
+end
+
+##############################
+# Example bijector: Identity #
+##############################
+
+struct Identity <: Bijector end
+(::Identity)(x) = x
+(::Inversed{Identity})(y) = y
+
+forward(::Identity, x) = (rv=x, logabsdetjac=zero(eltype(x)))
+
+logabsdetjac(::Identity, y) = zero(eltype(y))
+
+const IdentityBijector = Identity()
+
+###############################
+# Example: Logit and Logistic #
+###############################
+using StatsFuns: logit, logistic
+
+struct Logit{T<:Real} <: Bijector
+    a::T
+    b::T
+end
+
+(b::Logit)(x) = @. logit((x - b.a) / (b.b - b.a))
+(ib::Inversed{<:Logit{<:Real}})(y) = @. (ib.orig.b - ib.orig.a) * logistic(y) + ib.orig.a
+
+logabsdetjac(b::Logit{<:Real}, x) = @. - log((x - b.a) * (b.b - x) / (b.b - b.a))
+
+#############
+# Exp & Log #
+#############
+
+struct Exp <: Bijector end
+struct Log <: Bijector end
+
+(b::Log)(x) = @. log(x)
+(b::Exp)(y) = @. exp(y)
```

Comment:
We often need to apply this to just one of the dimensions. Any ideas on how to go about this?

Reply:
Me and @xukai92 were discussing this earlier, and we feel like batch computation is something we'll have to have a proper discussion about after this PR is merged. For now assume batch-computation is _not_ supported :/ The difficulty is that in a lot of cases it's ambiguous whether or not something is a batch. And we want to make sure that if we say we support batch-computation, then it's supported for _all_ bijectors, not just on a case-by-case basis.

I know you use it a lot though, so I'd suggest making a different branch where you try to accommodate batch-computation and then we can potentially use this when we're considering the approach we're going to take to support it:)

Discussion: https://github.com/TuringLang/Bijectors.jl/pull/27#discussion_r317863600

Reply:
@sharanry @torfjelde Let's experiment this feature in a different branch. I'm going to merge this PR soon.